### PR TITLE
Update build requirements

### PR DESCRIPTION
**Pull request by dveeden**
Mostly for CentOS 7:

- Add libev, libcurl, libgcrypt
- Update libncurses package name
- Remove bzr, autotools, autoconf
- Add sphinx-build

Fixes https://jira.percona.com/browse/PXB-2596

### CODE REVIEW

File: BUILD.XtraBackup.txt [hunk 1]
```diff
@@ -10,14 +10,13 @@ Percona XtraBackup from source. These might vary from system to system.
...
+  $ apt-get install build-essential flex bison cmake libaio-dev \
+    mysql-client libncurses-dev zlib1g-dev libgcrypt11-dev
 
-In RPM-based distributions, you need to:
+In RPM-based distribution like CentOS 7, you need to:
 
-  $ yum install cmake gcc gcc-c++ libaio libaio-devel automake autoconf bzr \
-    bison libtool ncurses5-devel
+  $ yum install cmake gcc gcc-c++ libaio libaio-devel bison ncurses-devel \
+    libgcrypt-devel libcurl-devel libev-devel python-sphinx
```
Comment (hrvojem) [inline BUILD.XtraBackup.txt]: missing vim-common, without it build will fail with:

> -- libev libraries found at: /usr/lib64/libev.so
> -- libev includes found at: /usr/include/libev
> CMake Error at storage/innobase/xtrabackup/src/CMakeLists.txt:31 (MESSAGE):
>   xxd not found.  Try to install vim-common.

File: BUILD.XtraBackup.txt [hunk 2]
```diff
@@ -10,14 +10,13 @@ Percona XtraBackup from source. These might vary from system to system.
...
 In Debian-based distributions, you need to:
 
-  $ apt-get install build-essential flex bison automake autoconf bzr \
-    libtool cmake libaio-dev mysql-client libncurses-dev zlib1g-dev \
-    libgcrypt11-dev
+  $ apt-get install build-essential flex bison cmake libaio-dev \
+    mysql-client libncurses-dev zlib1g-dev libgcrypt11-dev
 
-In RPM-based distributions, you need to:
+In RPM-based distribution like CentOS 7, you need to:
```
Comment (hrvojem) [inline BUILD.XtraBackup.txt]: s/CentOS 7/CentOS/

because all three (5/6/7) versions are currently supported

File: BUILD.XtraBackup.txt (hunk 1 above)
Comment (dveeden) [reply BUILD.XtraBackup.txt]: Fixed that. Also added libcurl and libev for debian.

File: BUILD.XtraBackup.txt (hunk 2 above)
Comment (dveeden) [reply BUILD.XtraBackup.txt]: Slightly changed that. The version number was because older versions have a different package name for ncurses.

**Comment by EvgeniyPatlan** [comment]
Thank you for your contribution! Before your contribution can be reviewed, we kindly ask you to sign our Contributor License Agreement (CLA).

**Comment by altmannmarcelo** [comment]
@EvgeniyPatlan / @hrvojem can you please check if this PR is still valid? Shall we merge it to 2.4 and 8.0 ?

**Comment by EvgeniyPatlan** [comment]
@altmannmarcelo I have rechecked this and seems dependencies are correct and we can merge this changes.
@dveeden thanks for your contribution!

**Review by EvgeniyPatlan — APPROVED** [review]

**Pull request merged**